app/views/contexts/summary/calculated_summary_block.py

```diff
@@ -74,9 +87,12 @@ def _build_link(self) -> str:
         return url_for(
             "questionnaire.block",
             block_id=self.id,
+            list_name=self._list_name,
+            list_item_id=self._list_item_id,
             return_to=self._return_to,
             return_to_answer_id=self.id,
             return_to_block_id=self._return_to_block_id,
+            return_to_list_item_id=self._return_to_list_item_id,
```

Comment:
If `return_to_list_item_id` is the same as `list_item_id` i.e we are changing an answer in the same repeating section then we shouldn't add this to the url, as it is not needed.

Reply:
Yeah that seems sensible, altered it here to only use the param if not in a repeating section, and made the change to router so we don't use it now inside the same repeat